johannes_mueller/entry_sphinx/Readme.md

```diff
@@ -0,0 +1,307 @@
+```
+docs/
+  ├─── make.bat
+  ├─── Makefile
+  ├─── conf.py
+  └─── index.rst
+```
+
+The `make.bat` allows you to generate html pages from the existing files, which is very handy. Create your first html pages using the command
+```make html```. If you navigate into `docs/_build/html` afterwards, you see a number of created html files. The most important of these files is `index.html`, which serves as your documentation's frontpage:
+
+![screenshot_1](https://user-images.githubusercontent.com/38459088/143234166-e048ebc5-903f-4eec-821c-afb6693318a0.JPG)
+
+This looks nice, but as you see, there's nothing on this page yet. Before we find our packages and modules on this page, we have to adjust a few settings in the configuration, which is done by making changes to `conf.py`.
+
+1. First, you'll have to change the directory where Sphinx looks for modules. At the top of `conf.py`, uncomment the following lines and change `sys.path.insert(0, os.path.abspath('.'))` into `sys.path.insert(0, os.path.abspath('../'))`.
+
+  ```Python
+  import os
+  import sys
+  sys.path.insert(0, os.path.abspath('../'))
+  ```
+  This tells Sphinx where to look for modules are classes to be documented.
+  
+2. Sphinx by itself is not able to parse docstrings of Python functions, but rather relies on extensions that take care of such things. In order to document functions or modules, we need to add the *autodoc* extension to Sphinx:
+```Python
+extensions = [
+  'sphinx.ext.autodoc',  # Parses docstrings
+]
+
+```
+3. We now need to tell Sphinx, that the docstrings of our modules, submodules and functions should actually be added to the generated documentation pages. For this, we need to make a few changes to `index.rst`, which ccurrently looks like this:
+
+```
+.. Very good package documentation master file, created by
+   sphinx-quickstart on Wed Nov 24 12:43:42 2021.
+   You can adapt this file completely to your liking, but it should at least
+   contain the root `toctree` directive.
+
+Welcome to Very good package's documentation!
+=============================================
+
+.. toctree::
+   :maxdepth: 2
+   :caption: Contents:
+
+
+
+Indices and tables
+==================
+
+* :ref:`genindex`
+* :ref:`modindex`
+* :ref:`search`
+```
+*Note*: This looks very familiar to the html index page from above, and `index.rst` is, in fact, its blueprint. These *.rst* files (so-called "stubs") to are used to tell Sphinx what should be documented. These stubs can be partly autogenerated but can and should partly be edited manually. These stub files can be used to fill html pages with documentation elements by adding so-callled *directives*. Directives in Sphinx are usually introduced with the `.. directive` syntax. For instance, in the above-example of `index.rst`, the table-of-contents directive (`.. toctree::`) is added with the options `:maxdepth: 2` and `:caption: Contents:`, but no entries have been added to this table of contents.
```

Comment:
`These *.rst* files (so-called "stubs") **to** are used to tell Sphinx what should be documented. ` I think this 'to' is unnecessary.